src/gui/search/searchsortmodel.cpp

```diff
@@ -47,8 +119,47 @@ bool SearchSortModel::lessThan(const QModelIndex &left, const QModelIndex &right
         if (Utils::String::naturalSort(vL.toString(), vR.toString(), res))
             return res;
 
-        return QSortFilterProxyModel::lessThan(left, right);
+        return base::lessThan(left, right);
+    }
+
+    return base::lessThan(left, right);
+}
+
+bool SearchSortModel::filterAcceptsRow(int source_row, const QModelIndex &source_parent) const
+{
+    if (m_isNameFilterEnabled && !m_searchTerm.isEmpty()) {
+        QString name = sourceModel()->data(sourceModel()->index(source_row, NAME)).toString();
+        for (const QString& word: m_searchTermWords) {
+            int i = name.indexOf(word, 0, Qt::CaseInsensitive);
+            if (i == -1) {
+                return false;
+            }
+        }
+    }
+
+    if (m_minSize > 0 || m_maxSize >= 0) {
+        qlonglong size = sourceModel()->data(sourceModel()->index(source_row, SIZE)).toLongLong();
+        if ((m_minSize > 0 && size < m_minSize)
+        ||  (m_maxSize > 0 && size > m_maxSize)) {
+            return false;
+        }
+    }
+
+    if (m_minSeeds > 0 || m_maxSeeds >= 0) {
+        int seeds = sourceModel()->data(sourceModel()->index(source_row, SEEDS)).toInt();
+        if ((m_minSeeds > 0 && seeds < m_minSeeds)
+        ||  (m_maxSeeds > 0 && seeds > m_maxSeeds)) {
```

Comment:
The same answer. I'm not going to argue about this. We have such rules. No need to break them. If they don't like me, can try to make a proposal with the best rules. But not here. This is a separate issue. I also do not like some rules. But the absence of any rules it is a great evil which makes the code unreadable. Let's never return to this question in this way.